jolt-core/src/jolt/vm/rv32i_vm.rs

```diff
@@ -321,11 +321,151 @@ mod tests {
             circuit_flags,
             &mut transcript,
             &mut random_tape,
+            3,
             JoltPaths::witness_generator_path("fibonacci"),
             JoltPaths::r1cs_path("fibonacci")
         );
     }
 
+    #[test]
+    fn hash_rest() {
+        use common::{path::JoltPaths, serializable::Serializable, ELFInstruction};
+        compiler::cached_compile_example("hash");
+
+        let trace_location = JoltPaths::trace_path("hash");
+        let loaded_trace: Vec<common::RVTraceRow> =
+            Vec::<common::RVTraceRow>::deserialize_from_file(&trace_location)
+                .expect("deserialization failed");
+        let bytecode_location = JoltPaths::bytecode_path("hash");
+        let bytecode = Vec::<ELFInstruction>::deserialize_from_file(&bytecode_location)
+            .expect("deserialization failed");
+        let bytecode_rows = bytecode.iter().map(ELFRow::from).collect();
+
+        let converted_trace: Vec<RVTraceRow> = loaded_trace
+            .into_iter()
+            .map(|common| RVTraceRow::from_common(common))
+            .collect();
+
+        let bytecode_trace: Vec<ELFRow> = converted_trace
+            .iter()
+            .map(|row| row.to_bytecode_trace())
+            .collect();
+
+        let instructions: Vec<RV32I> = converted_trace
+            .clone()
+            .into_iter()
+            .flat_map(|row| row.to_jolt_instructions())
+            .collect();
+
+        // Emulator sets register 0xb to 0x1020 upon initialization for some reason,
+        // something about Linux boot requiring it...
+        let mut memory_trace: Vec<MemoryOp> = vec![MemoryOp::Write(11, 4128)];
```

Comment:
If we do this every time we should likely move into a default function called "gen_memory_trace" or something of that nature. (Can deal with in later PR, add a TODO here)